web/src/queries/storage/config-model.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { useMutation, useQuery, useQueryClient, useSuspenseQuery } from "@tanstack/react-query";
+import { fetchConfigModel, setConfigModel } from "~/api/storage";
+import { configModel } from "~/api/storage/types";
+import { QueryHookOptions } from "~/types/queries";
+import { SpacePolicyAction } from "~/types/storage";
+
+function findDrive(model: configModel.Config, driveName: string): configModel.Drive | undefined {
+  const drives = model.drives || [];
+  return drives.find((d) => d.name === driveName);
+}
+
+// TODO: add a second drive if needed (e.g., reusing a partition).
+function changeDrive(model: configModel.Config, driveName: string, newDriveName: string) {
+  const drive = findDrive(model, driveName);
+  if (drive === undefined) return;
+
+  drive.name = newDriveName;
+  if (drive.spacePolicy === "custom") drive.spacePolicy = "keep";
```

Comment:
Why "keep" here without even a TODO? I would have expected to use the default product policy.

Reply:
As discussed in a chat, the are going to follow this approach:

* Use "keep" for a device that is added as effect of using it as boot device.
* Honor the default policy defined by the product in other cases.

@ancorgs Applying this change would be easier to do in the follow-up PR https://github.com/agama-project/agama/pull/1808. This part of the code has changed a bit. Do you mind if we keep this as it is with a TODO?

Reply:
No problem.